IA/IA_upload.py

```diff
@@ -1,109 +1,84 @@
-
-    await asyncio.gather(*tasks)
-
-
-async def upload(bucket_name: str, filename: str, file_content: bytes):
-    headers = {
-        'authorization': 'LOW {}:{}'.format(IA_ACCESS_KEY, IA_SECRET_KEY),
-        'x-amz-auto-make-bucket': '1',
-        'Content-Type': 'application/octet-stream',
-        'x-archive-meta01-collection': OSF_COLLECTION_NAME,
-    }
-    url = f'{IA_URL}/{bucket_name}/{filename}'
-    resp = put_with_retry(f'http://{url}', headers=headers, data=file_content, retry_on=(429, 503))
 
-    if resp.status_code != 200:
-        error_json = dict(xmltodict.parse(resp.content))
-        raise requests.exceptions.HTTPError(error_json)
+    ia_upload(ia_item, directory_path + '/', session)
+    upload_metadata(ia_item, guid, parent, session)
 
 
-async def chunked_upload(bucket_name: str, filename: str, file_content: bytes):
-    conn = boto.connect_s3(
-        IA_ACCESS_KEY,
-        IA_SECRET_KEY,
-        host=IA_URL,
-        is_secure=False,
-        calling_format=OrdinaryCallingFormat(),
-    )
-    bucket = conn.lookup(bucket_name)
-    if bucket is None:
-        bucket = conn.create_bucket(
-            bucket_name,
-            headers={'x-archive-meta01-collection': OSF_COLLECTION_NAME}
-        )
-
-    initiate_multipart_with_retry = sleep_and_retry_on(
-        bucket.initiate_multipart_upload,
-        retry_on=boto.exception.S3ResponseError
-    )
+def upload_metadata(ia_item: internetarchive.item,
+                    guid: str, directory: str, session: internetarchive.session):
 
-    mp = initiate_multipart_with_retry(filename)
+    node_path = os.path.join(HERE, directory, 'data', guid, 'node', f'{guid}.json')
+    with open(node_path, 'r') as f:
+        node_json = json.loads(f.read())['attributes']
 
-    tasks = []
-    chunks = [file_content[i:i + CHUNK_SIZE] for i in range(0, len(file_content), CHUNK_SIZE)]
+    xml_path = os.path.join(HERE, directory, 'data', 'datacite.xml')
+    xml_data = minidom.parse(xml_path)
+    creators = xml_data.getElementsByTagName('creatorName')
 
-    for i, chunk in enumerate(chunks):
-        mp = mp_from_ids(mp.id, filename, bucket)
-        upload_part_from_file = asyncio.coroutine(mp.upload_part_from_file)
-        tasks.append(asyncio.ensure_future(upload_part_from_file(BytesIO(chunk), i + 1)))
+    date_string = node_json['date_created']
+    date_string = date_string.partition('.')[0]
+    date_time = datetime.strptime(date_string, "%Y-%m-%dT%H:%M:%S")
 
-    await asyncio.gather(*tasks)
+    metadata_dict = dict(
+        title=node_json['title'],
+        description=node_json['description'],
+        creator=creators[0].firstChild.data,
+        date=date_time.strftime("%Y-%m-%d"),
+        subjects=', '.join(node_json['subjects']),
+        contributor='Center for Open Science',
+    )
+
+    article_doi = node_json['article_doi']
+    metadata_dict['external-identifier'] = f'urn:doi:{article_doi}'
+    ia_item.modify_metadata(metadata_dict)
```

Comment:
I kept the upload metadata there since the docs seemed to say that if the metadata already existed, it would not be updated by the upload function (not sure if this is the case for get_item). I'm not sure if we need to update the metadata each time we update the archive, but I assume we would.